apps/server/src/services/institutionSyncer.test.ts

```diff
@@ -0,0 +1,135 @@
+import { http, HttpResponse } from 'msw'
+import { clearIntervalAsync } from 'set-interval-async'
+import { server } from '../test/testServer'
+import * as syncerModule from './institutionSyncer'
+import {
+  loadCachedInstitutions,
+  setInstitutionSyncSchedule
+} from './institutionSyncer'
+
+import { writeFileSync } from 'fs'
+import config from '../config'
+import * as logger from '../infra/logger'
+import { elasticSearchInstitutionData } from '../test/testData/institution'
+import * as auth0Module from './auth0Service'
+import * as esModule from './ElasticSearchClient'
+import { getInstitutionFilePath } from './ElasticSearchClient'
+import * as redis from './storageClient/redis'
+
+jest.mock('fs')
+
+describe('setInstitutionSyncSchedule', () => {
+  beforeEach(() => {
+    jest.useFakeTimers()
+  })
+
+  afterEach(() => {
+    jest.useRealTimers()
+    jest.restoreAllMocks()
+  })
+
+  it.only('should call sync over the given interval', async () => {
+    const loadCachedSpy = jest
+      .spyOn(syncerModule, 'loadCachedInstitutions')
+      .mockResolvedValue()
+
+    const poller = await setInstitutionSyncSchedule(0.25)
+
+    expect(loadCachedSpy).not.toHaveBeenCalled()
+
+    jest.advanceTimersByTime(15000)
+    await Promise.resolve()
+    expect(loadCachedSpy).toHaveBeenCalledTimes(1)
+
+    await Promise.resolve()
```

Comment:
What are all these Promise.resolves for?

Reply:
This was a weird thing I discovered when I was trying to test. I was not getting the results I was expecting when I was doing `jest.advanceTimersByTime` and I think it's because of the nature of the async interval, which won't trigger the function until the previous one is resolved. So I think when I add Promise.resolve(), the test advances the time, and then waits for the async function to resolve. In short I think Promise.resolve() tells the test to slow down for a sec. Having two right after each other is probably not necessary though, I'll remove that.